session/txninfo/summary.go

```diff
@@ -120,28 +194,39 @@ func (recorder *TrxHistoryRecorder) DumpTrxSummary() [][]types.Datum {
 	return recorder.summaries.dumpTrxSummary()
 }
 
+// DumpTrxIDDigests dumps the transaction id-digest mapping to Datum for displaying in `TRX_ID_DIGEST` table.
+func (recorder *TrxHistoryRecorder) DumpTrxIDDigests() [][]types.Datum {
+	recorder.mu.Lock()
+	defer recorder.mu.Unlock()
+	return recorder.idDigests.dump()
+}
+
 // OnTrxEnd should be called when a transaction ends, ie. leaves `TIDB_TRX` table.
 func (recorder *TrxHistoryRecorder) OnTrxEnd(info *TxnInfo) {
 	now := time.Now()
 	startTime := time.Unix(0, oracle.ExtractPhysical(info.StartTS)*1e6)
-	if now.Sub(startTime) < recorder.minDuration {
+	duration := now.Sub(startTime)
+	if duration < recorder.minDuration {
```

Comment:
Not related to this PR, it's a bit strange to calculate the diff between a system time and a TSO time.